### PR TITLE
Feature/227 noxgenerators odata get with multiple primary keys

### PR DESCRIPTION
**Pull request by anna-naumova25**
Added multipleIdsType entity to SampleWebApp to illustrate how Get by multiple keys work. Initially it should be done in scope of AllNoxTypes, but while I worked on the feature, AllNoxTypes moved to databaseNumber key which cannot be used twice per model. So if you want me to move/change using of compound primary keys, please let me know.

### CODE REVIEW

File: src/Nox.Generator/Application/Dto/EntityDto.template.cs [hunk 1]
```diff
@@ -17,8 +21,10 @@ namespace {{codeGeneratorState.ApplicationNameSpace}}.Dto;
 /// {{entity.Description}}.
 /// </summary>
 [AutoMap(typeof({{entity.Name}}CreateDto))]
+[PrimaryKey({{primaryKeyParams}})]
```
Comment (rochar) [inline src/Nox.Generator/Application/Dto/EntityDto.template.cs:24]: this change should not be necessary, the model and the primary keys are being set by the Databaseconfiguration

I would expect changes only on the controller level

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, one comment to discuss

**Comment by rochar** [comment]
can we also create a test on Nox.Generators asserting the result of a controller with multiple keys

File: src/Nox.Generator/Application/Dto/EntityDto.template.cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Application/Dto/EntityDto.template.cs:24]: Unfortunately, the following exception is thrown on any API call if this line of code is removed:

{
  "statusCode": 500,
  "Message": "The entity type 'MultipleIdsTypeDto' has multiple properties with the [Key] attribute. Composite primary keys configured by placing the [PrimaryKey] attribute on the entity type class, or by using 'HasKey' in 'OnModelCreating'."
} 

Even considering that we have HasKey, it still doesn't work without PrimaryKey

Comment (anna-naumova25) [reply src/Nox.Generator/Application/Dto/EntityDto.template.cs]: As discussed, updated to HasKey

**Comment by anna-naumova25** [comment]
Added controller test to Nox.Generator.

Updated AllNoxTypes to using compound keys.

Will merge with main once I fix local build.

**Comment by rochar** [comment]
@anna-naumova25  I see the deleted migration files, I do no see the new ones?

when @alexandervlasenko approves it, please merge and delete the branch

**Review by alexandervlasenko — APPROVED** [review]
Please, push the updated db migrations. Otherwise everything is fine.

File: src/Nox.Generator/Application/Queries/ByIdQuery.template.cs [hunk 2]
```diff
@@ -24,7 +24,16 @@ public class Get{{entity.Name}}ByIdQueryHandler: IRequestHandler<Get{{entity.Nam
     {    
         var item = DataDbContext.{{entity.PluralName}}
             .AsNoTracking()
-            .SingleOrDefault(r => {{if (entity.Persistence?.IsVersioned ?? true)}}!(r.Deleted == true) && {{end}}r.{{entity.Keys[0].Name}}.Equals(request.key));            
+            .SingleOrDefault(r =>                  
+            {{- for key in entity.Keys }}
+                r.{{key.Name}}.Equals(request.key{{key.Name}}) && 
+            {{- end -}}
+            {{- if (entity.Persistence?.IsVersioned ?? true)}}
+                !(r.Deleted == true)
```
[the remark this answers is not in the document]
Comment (rochar) [reply src/Nox.Generator/Application/Queries/ByIdQuery.template.cs:32]: since Delete is nullable that would create a wrong query, ignoring null values

**Pull request merged**